pkg/controller/v1beta1/inferenceservice/components/predictor.go

```diff
@@ -189,9 +192,49 @@ func (p *Predictor) Reconcile(ctx context.Context, isvc *v1beta1.InferenceServic
 	} else {
 		var err error
 		podLabelKey = constants.RevisionLabel
+
 		if kstatus, err = p.reconcileKnativeDeployment(ctx, isvc, &objectMeta, &podSpec); err != nil {
 			return ctrl.Result{}, err
 		}
+		if p.forceStopRuntime {
+			// Exit early if we have already set the status to stopped
+			existing_stopped_condition := isvc.Status.GetCondition(v1beta1.Stopped)
+			if existing_stopped_condition != nil && existing_stopped_condition.Status == corev1.ConditionTrue {
+				return ctrl.Result{}, nil
+			}
+
+			// Clear all statuses
+			p.Log.Info("Clearing ISVC status")
+			isvc.Status = v1beta1.InferenceServiceStatus{}
+
+			// Set the ready conditions to false
```

Comment:
Could you please add more information why we add predictor_ready and ingress_ready conditions here?